src/components/timer/timer.tsx

```diff
@@ -0,0 +1,37 @@
+import { useEffect } from "react";
+import { useState } from "react";
+import { Time } from "./timer.types";
+
+export default function Timer({
+  minSec,
+  setTimeUp,
+}: {
+  minSec: Time;
+  setTimeUp: React.Dispatch<React.SetStateAction<boolean>>;
+}) {
+  //const [timer, setTimer] = useState(0);
+  const { minutes, seconds } = minSec;
+  const [[mins, secs], setTime] = useState([minutes, seconds]);
+
+  const tick = () => {
+    if (mins === 0 && secs === 0) {
+      setTimeUp(true);
+    }
+    if (secs === 0) {
+      setTime([mins - 1, 59]);
+    } else {
+      setTime([mins, secs - 1]);
+    }
+  };
```

Comment:
Change timing update condition as suggested
```suggestion    
    if (secs === 0) {
           if (mins === 0) {
                  return setTimeUp(true);
       }
      return setTime([mins - 1, 59]);
    }
    setTime([mins, secs - 1]);
  };
```

Reply:
this seems better thanks, fixed